internal/ingress/annotations/annotations.go

```diff
@@ -49,39 +57,52 @@ const (
 	DefaultIngressClass = "kong"
 )
 
-func validIngress(ingressAnnotationValue, ingressClass string, skipClasslessIngress bool) bool {
-	// we have 2 valid combinations
-	// 1 - ingress with default class | blank annotation on ingress
-	// 2 - ingress with specific class | same annotation on ingress
-	//
-	// and 2 invalid combinations
-	// 3 - ingress with default class | fixed annotation on ingress
-	// 4 - ingress with specific class | different annotation on ingress
-	if ingressAnnotationValue == "" && !skipClasslessIngress && ingressClass == DefaultIngressClass {
+func validIngress(ingressAnnotationValue, ingressClass string, handling ClassMatching) bool {
+	switch handling {
+	case IgnoreClassMatch:
+		// class is not considered at all. any value, even a mismatch, is valid.
+		// may want to consider warning on mismatches, since while valid, they probably indicate a conflict with user
+		// intent, and probably should be something users clear for good config hygiene
 		return true
+	case ExactOrEmptyClassMatch:
+		// aka lazy. exact match desired, but empty permitted
+		if ingressAnnotationValue == "" || ingressAnnotationValue == ingressClass {
+			return true
+		} // no alternative case, since everything else should be a mismatch
```

Comment:
You don't need an `if` here. Simply `return condition1 || condition 2` should suffice.